pkg/kubectl/cmd/set/set_data.go

```diff
@@ -0,0 +1,291 @@
+				return err
+			}
+			continue
+		}
+
+		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, types.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			allErrs = append(allErrs, fmt.Errorf("failed to patch data to : %v\n", err))
+			continue
+		}
+		info.Refresh(obj, true)
+
+		if len(o.Output) > 0 {
+			if err := cmdutil.PrintObject(o.Cmd, info.AsVersioned(), o.Out); err != nil {
+				return err
+			}
+			continue
+		}
+		cmdutil.PrintSuccess(o.ShortOutput, o.Out, info.Object, o.DryRun, "data updated")
+	}
+	return utilerrors.NewAggregate(allErrs)
+}
+
+func getData(objType string, obj interface{}, fileSources, literalSources []string, envFileSource string) error {
+	if objType == "secret" {
+		if len(fileSources) > 0 {
+			if err := kubectl.HandleFromFileSources(obj.(*v1.Secret), fileSources); err != nil {
+				return err
+			}
+		}
+		if len(literalSources) > 0 {
+			if err := kubectl.HandleFromLiteralSources(obj.(*v1.Secret), literalSources); err != nil {
+				return err
+			}
+		}
+		if len(envFileSource) > 0 {
+			if err := kubectl.HandleFromEnvFileSource(obj.(*v1.Secret), envFileSource); err != nil {
+				return err
+			}
+		}
+	} else if objType == "configmap" {
+		if len(fileSources) > 0 {
+			if err := kubectl.HandleConfigMapFromFileSources(obj.(*v1.ConfigMap), fileSources); err != nil {
+				return err
+			}
+		}
+		if len(literalSources) > 0 {
+			if err := kubectl.HandleConfigMapFromLiteralSources(obj.(*v1.ConfigMap), literalSources); err != nil {
+				return err
+			}
+		}
+		if len(envFileSource) > 0 {
+			if err := kubectl.HandleConfigMapFromEnvFileSource(obj.(*v1.ConfigMap), envFileSource); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (o *DataOptions) updateDataForObject(obj runtime.Object, fn func(string, runtime.Object, interface{}) (bool, map[string]string, map[string][]byte)) (bool, error) {
+	switch t := obj.(type) {
+	case *core.Secret:
+		secretObj := &v1.Secret{}
+		secretObj.Data = map[string][]byte{}
+		err := getData("secret", secretObj, o.FileSources, o.LiteralSources, o.EnvFileSource)
+		if err != nil {
+			return false, err
+		}
+		transformed, _, bindata := fn("secret", t, secretObj.Data)
+		t.Data = bindata
+		return transformed, nil
+	case *core.ConfigMap:
+		configMapObj := &v1.ConfigMap{}
+		configMapObj.Data = map[string]string{}
+		configMapObj.BinaryData = map[string][]byte{}
+		err := getData("configmap", configMapObj, o.FileSources, o.LiteralSources, o.EnvFileSource)
+		if err != nil {
+			return false, err
+		}
+		transformed, data, bindata := fn("configmap", t, configMapObj.Data)
+		t.Data = data
+		t.BinaryData = bindata
+		return transformed, nil
+	default:
+		return false, fmt.Errorf("setting data is only supported for Secrets and ConfigMaps")
+	}
+}
+
+func addData(objType string, obj runtime.Object, target interface{}) (bool, map[string]string, map[string][]byte) {
```

Comment:
The `objType` is redundant and can only lead to inconsistency. You shouldn't need it.